overlays/overlay.nix

```diff
@@ -14,4 +15,15 @@ final: prev:
       );
 
   polykey-cli = inputs.polykey-cli.packages.x86_64-linux.default;
+
+  gitignore = excludes: src:
+    let
+      gitignoreContents = builtins.readFile (src + "/.gitignore");
+      extraRules = builtins.concatStringsSep "\n" excludes;
+      srcIgnored = gitignoreFilterWith {
+        basePath = src;
+        extraRules = "${gitignoreContents}\n${extraRules}";
+      };
+    in
+      builtins.filterSource (path: type: srcIgnored path type) src;
```

Comment:
I don't understand why this is necessary? What is the purpose of this?

Reply:
I added this to make a wrapper around hercules ci-s gitignore, as it doesn't allow for additional rules by default. This ensures that the syntax we have will be kept as-is like

```
src = gitignore [ 'extra-files-to-ignore' ] ./.
```